src/software/ai/navigator/obstacle/obstacle_factory.h

```diff
@@ -94,29 +94,44 @@ class ObstacleFactory
      *
      * @return rectangular obstacle
      */
-    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle);
+    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle) const;
 
    private:
     std::shared_ptr<const ObstacleFactoryConfig> config;
-    double shape_expansion_amount;
+    double obstacle_expansion_amount;
 
     /**
-     * Create circle obstacle directly from a Circle
-     * Note: this helper function does not add a robot radius
+     * Returns an obstacle for the shape expanded on all sides to account for the size of
+     * the robot
      *
-     * @param circle Circle
+     * @param The shape to expand
      *
-     * @return circle shaped obstacle
+     * @return expanded ObstaclePtr
      */
-    ObstaclePtr createObstacle(const Circle &circle);
+    ObstaclePtr expandForRobotSize(const Circle &circle) const;
+    ObstaclePtr expandForRobotSize(const Polygon &polygon) const;
```

Comment:
Sorry if I'm totally missing something here, but I'm a bit confused about what the intent of these `expandForRobotSize` functions is. Could you maybe re-explain at a high level? Hoping I can help come up with a name that maybe better conveys the intent.

Reply:
The idea is to take a shape, add a margin around the shape of size `ROBOT_MAX_RADIUS * dynamic param value`, then construct and return an `Obstacle` of with that bigger shape

Reply:
So what sort of obstacle are we supposed to pass into it? An obstacle that is roughly the same size as the robot? Something else?

Reply:
Let's take an example of an obstacle that represents `HALF_METER_AROUND_BALL`. So pass in a `Circle( centre = ball_pos, radius = 0.5)` to `expandForRobotSize`. Then, it expands the circle to account for the size of the robot so the `Obstacle` will be a `Circle( centre = ball_pos, radius = 0.617)`. The function then returns that bigger `Obstacle`. This simplification is useful for navigation, because we can navigate with a 0-width path, instead of with a robot diameter-width path

Reply:
Oh, I think I get what this does now. Perhaps `inflateByRobotRadius` would be a more descriptive name?

Reply:
(apologies that this is dragging out a bit, let me know if you want to hop on a call for this, or just message me on slack so we can iterate quicker)